lib/omniauth/strategies/shopify.rb

```diff
@@ -49,7 +49,7 @@ def valid_signature?
       def valid_scope?(token)
         params = options.authorize_params.merge(options_for("authorize"))
         return false unless token && params[:scope] && token['scope']
-        (params[:scope].split(',').sort == token['scope'].split(',').sort)
+        (params[:scope].delete(' ').split(',').sort == token['scope'].delete(' ').split(',').sort)
```

Comment:
I should have captured the session... I didn't expect it to mysteriously stop being an issue. It was whichever one was coming back from the server :). When the issue appeared, my params had no spaces, I had cleared cookies, reset my dev db, and hard reset my browser cache, and quit and restarted my browser, just to be certain. After reviewing documentation, I ended up trying it every which way, thinking it was picky. Apparently it isn't -- the user entered params seem to be cleaned up. This is theoretically overkill, but shouldn't do any harm to be better safe than sorry.